drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -121,6 +132,30 @@ local function temperature_attr_handler(driver, device, ib, response)
   end
 end
 
+local temp_attr_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
+    local temp = ib.data.value / 100.0
+    local unit = "C"
+    set_field_for_endpoint(device, TEMP_BOUND_RECEIVED..minOrMax, ib.endpoint_id, temp)
+    local min = get_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MIN, ib.endpoint_id)
+    local max = get_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MAX, ib.endpoint_id)
+    -- Only emit the capability for RPC version >= 5 (unit conversion for
+    -- temperature range capability is only supported for RPC >= 5)
+    if version.rpc >= 5 and min ~= nil and max ~= nil then
+      if min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureMeasurement.temperatureRange({ value = { minimum = min, maximum = max }, unit = unit }))
+        set_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MIN, ib.endpoint_id, nil)
+        set_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MAX, ib.endpoint_id, nil)
+      else
+        device.log.warn_with({hub_logs = true}, string.format("Device reported a min temperature %d that is not lower than the reported max temperature %d", min, max))
```

Comment:
I think it would still be useful to report this log even for RPC version < 5 since these values will still be used. The only thing that really needs to be gated by RPC version is emitting the capability event.

Reply:
Changed as suggested in the latest commit!